sequence_processing_pipeline/tests/test_Pipeline.py

```diff
@@ -1,72 +1,148 @@
+
+    def test_creation(self):
+        # Pipeline should assert due to config_file
+        with self.assertRaises(PipelineError):
+            Pipeline(self.bad_config_file, self.good_run_id)
+
+        # Pipeline should assert due to an invalid config file path.
+        with self.assertRaises(PipelineError):
+            Pipeline(self.invalid_config_file, self.good_run_id)
+
+        # Pipeline should assert on config_file = None
+        with self.assertRaises(PipelineError):
+            Pipeline(None, self.good_run_id)
+
+        # Pipeline should assert due to invalid_run_id
+        with self.assertRaises(PipelineError):
+            Pipeline(self.good_config_file, self.invalid_run_id)
+
+        # Pipeline should assert on run_id = None
+        with self.assertRaises(PipelineError):
+            Pipeline(self.good_config_file, None)
+
+        with open(join('sequence_processing_pipeline', 'configuration.json'),
+                  'r') as f:
+            cfg = json.load(f)
+            with self.assertRaises(PipelineError):
+                cpy_cfg = deepcopy(cfg)
+                cpy_cfg['configuration']['pipeline']['younger_than'] = -1
+                logging.debug(json.dumps(cpy_cfg, indent=2))
+                Pipeline(self.good_config_file, self.good_run_id,
+                         config_dict=cpy_cfg)
+            with self.assertRaises(PipelineError):
+                cpy_cfg = deepcopy(cfg)
+                cpy_cfg['configuration']['pipeline']['older_than'] = -1
+                Pipeline(self.good_config_file, self.good_run_id,
+                         config_dict=cpy_cfg)
+            with self.assertRaises(PipelineError):
+                cpy_cfg = deepcopy(cfg)
+                cpy_cfg['configuration']['pipeline']['younger_than'] = 20
+                cpy_cfg['configuration']['pipeline']['older_than'] = 30
+                Pipeline(self.good_config_file, self.good_run_id,
+                         config_dict=cpy_cfg)
+
+    def test_find_bcl_directories(self):
+        pipeline = Pipeline(self.good_config_file, self.good_run_id)
+        obs = pipeline.find_bcl_directories(self.data_directory)
+        # there are a number of directories under 'data'. Only
+        # 'sample-sequence-directory' has the layout expected of a
+        # run-directory, however.
+        exp = ["sequence_processing_pipeline/tests/data/"
+               "sample-sequence-directory"]
+        self.assertEqual(obs, exp)
+
+    def test_filter_directories_for_time(self):
+        # get a good configuration dict from file.
+        with open(join('sequence_processing_pipeline', 'configuration.json'),
+                  'r') as f:
+            cfg = json.load(f)
+            # set a range that sample-sequence-directory's timestamps must
+            # fall within: between 1 and 2 hours.
+            cfg['configuration']['pipeline']['younger_than'] = 2
+            cfg['configuration']['pipeline']['older_than'] = 1
+
+            # Test 1
```

Comment:
If there are different test cases to need to number them, perhaps worth creating individual tests for each one, no? Either way, `Test 1` is not clear on what's being tested, could you make it clearer?